activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -560,7 +560,7 @@ def fetch_type_metadata(column_name, sql_type, oid, fmod)
           simple_type = SqlTypeMetadata.new(
             sql_type: sql_type,
             type: cast_type.type,
-            limit: cast_type.limit,
+            limit: cast_type.try(:subtype).try(:limit) || cast_type.limit,
```

Comment:
I think `(cast_type.subtype ? cast_type.subtype.limit : cast_type_limit)` is preferable over `try` here.